cmd/participation/main.go

```diff
@@ -0,0 +1,169 @@
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	clientCertPool := x509.NewCertPool()
+	caCert, err := ioutil.ReadFile(filepath.Join(tlsDir, "ca.crt"))
+	if err != nil {
+		panic(err)
+	}
+	clientCertPool.AppendCertsFromPEM(caCert)
+
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				Certificates: []tls.Certificate{clientCert},
+				RootCAs:      clientCertPool,
+			},
+		},
+	}
+}
+
+func getBodyBytes(client *http.Client, url string) ([]byte, error) {
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, nil
+	}
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading http response body")
+	}
+	resp.Body.Close()
+
+	return bodyBytes, nil
+}
+
+func printResponseAsJSON(bodyBytes []byte, response interface{}, out io.Writer) error {
+	err := json.Unmarshal(bodyBytes, response)
+	if err != nil {
+		return errors.Wrapf(err, "unmarshaling body bytes into response type: %T", response)
+	}
+
+	bytes, err := json.MarshalIndent(response, "", "\t")
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal output")
+	}
+
+	fmt.Fprintf(out, "%s\n", string(bytes))
+
+	return nil
+}
+
+func validateBlockChannelID(blockBytes []byte, channelID string) error {
+	block := &common.Block{}
+	err := proto.Unmarshal(blockBytes, block)
+	if err != nil {
+		return errors.Wrap(err, "unmarshaling block")
+	}
+
+	blockChannelID, err := protoutil.GetChannelIDFromBlock(block)
+	if err != nil {
+		return err
+	}
+
+	// quick sanity check that the orderer admin is joining
+	// the channel they think they're joining.
+	// NOTE: we could remove this and the channelID flag from
+	// the join CLI command but I think it may be beneficial to
+	// keep this around since otherwise they'd need to use
+	// configtxgen and manually look through the block output
+	// to confirm the channelID matches what they expect
+	if channelID != blockChannelID {
+		return errors.Errorf("Specified channelID %s does not match channel ID %s in join block", channelID, blockChannelID)
+	}
+
+	return nil
+}
+
+func main() {
+	var osn, channelID, tlsDir, configBlockPath string
+
+	listCmd := flag.NewFlagSet("list", flag.ExitOnError)
+	listCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	listCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
+	listCmd.StringVar(&channelID, "channelID", "", "Channel ID")
+
+	removeCmd := flag.NewFlagSet("remove", flag.ExitOnError)
+	removeCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	removeCmd.StringVar(&channelID, "channelID", "", "Channel ID")
+	removeCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
+
+	joinCmd := flag.NewFlagSet("join", flag.ExitOnError)
+	joinCmd.StringVar(&osn, "orderer", "", "Ordering service endpoint")
+	joinCmd.StringVar(&tlsDir, "tlsDir", "", "Path to directory containing TLS materials")
```

Comment:
The contents of this directory should be validated before we proceed with the invocations.  Going back to a previous comment, we should do all of our parsing and setup early and in a consolidated place.